typescript/sdk/src/deploy/validation.ts

```diff
@@ -0,0 +1,140 @@
+ *
+ * @throws OwnerValidationError - When any owner is inactive/error/skipped
+ */
+export async function validateWarpDeployOwners(
+  warpConfig: WarpRouteDeployConfigMailboxRequired,
+  multiProvider: MultiProvider,
+): Promise<void> {
+  logger.info('Starting owner validation for warp deployment');
+
+  // Extract owners from configuration
+  const ownersByChain = extractOwnersFromConfig(warpConfig);
+
+  logger.debug('Extracted owners for validation', {
+    ownersByChain: Object.fromEntries(
+      objMapEntries(ownersByChain, (chain, owners) => [chain, owners.length]),
+    ),
+  });
+
+  // Validate all owners in parallel per chain, sequential per owner to avoid rate limiting
+  const invalidOwners: OwnerValidationResult[] = [];
+
+  for (const [chain, owners] of Object.entries(ownersByChain)) {
+    logger.debug(`Validating ${owners.length} owners for chain ${chain}`);
+
+    for (const owner of owners) {
+      const result = await validateOwnerActivity(chain, owner, multiProvider);
+
+      // Check if status is invalid (not Active or GnosisSafe)
+      if (
+        result.status !== OwnerStatus.Active &&
+        result.status !== OwnerStatus.GnosisSafe
+      ) {
+        invalidOwners.push(result);
+        logger.warn('Invalid owner detected', {
+          chain,
+          address: owner,
+          status: result.status,
+        });
+      }
+    }
+  }
+
+  if (invalidOwners.length > 0) {
+    const errorMessage = `Owner validation failed: Found ${invalidOwners.length} invalid owner(s). Invalid owners:\n${invalidOwners.map(
+      (o) => `\n- Chain: ${o.chain}, Owner: ${o.address}, Status: ${o.status}`,
+    )}`;
+
+    throw new OwnerValidationError(invalidOwners, errorMessage);
+  }
+
+  logger.info('Owner validation completed successfully');
+}
+
+/**
+ * Extracts all owner addresses from warp deployment configuration.
+ * Handles owner overrides and proxy admin owners.
+ */
+export function extractOwnersFromConfig(
+  warpConfig: WarpRouteDeployConfigMailboxRequired,
+): Record<ChainName, Address[]> {
+  return objMap(warpConfig, (chain, config) => {
+    const owners: Address[] = [];
+
+    // 1. Primary owner - use ownerOverrides[chain] if present, otherwise config.owner
+    const primaryOwner = config.ownerOverrides?.[chain] || config.owner;
+    if (primaryOwner) {
+      owners.push(primaryOwner);
+    }
+
+    // 2. Proxy admin owner (if proxy deployment configured)
+    if (config.proxyAdmin?.owner) {
+      owners.push(config.proxyAdmin.owner);
+    }
+
+    return [...new Set(owners)];
+  });
+}
+
+/**
+ * Validates activity status of a single owner address on specific chain.
+ * Uses EvmERC20WarpRouteReader.validateOwnerAddress() for direct owner validation.
+ */
+export async function validateOwnerActivity(
+  chain: ChainName,
+  address: Address,
+  multiProvider: MultiProvider,
+): Promise<OwnerValidationResult> {
+  const reader = new EvmERC20WarpRouteReader(multiProvider, chain);
```

Comment:
if we extract the method we can avoid doing this here and just call the function